models/logs.js

```diff
@@ -35,15 +36,17 @@ const addLog = async (type, meta, body) => {
  */
 const fetchLogs = async (query, param) => {
   try {
+    const { dev, ...remainingQuery } = query;
     let call = logsModel.where("type", "==", param);
-    Object.keys(query).forEach((key) => {
+    Object.keys(remainingQuery).forEach((key) => {
       // eslint-disable-next-line security/detect-object-injection
       if (key !== "limit" && key !== "lastDocId") {
-        call = call.where(key, "==", query[key]);
+        // eslint-disable-next-line security/detect-object-injection
+        call = call.where(key, "==", remainingQuery[key]);
```

Comment:
why are we disabling  the lint here?

Reply:
If I remove this comment then it throws a warning 
``` Function Call Object Injection Sinkeslint(security/detect-object-injection)``` which states stands for 
> This rule is designed to catch potential security vulnerabilities related to object injection attacks. Object injection is a type of attack where an attacker can manipulate an object's properties or methods in order to execute malicious code.

In simple terms, it is saying that it is prone to injection attacks where the user can send any kind of query param that may not be allowed as we are directly looping over passed queries without verifying the keys sent from the front end.

Both these codes have the same vulnerability. But this API is only accessible by the super user. Maybe that is why it was never fixed.
```
    Object.keys(query).forEach((key) => {   // Previously implemented
    Object.keys(remainingQuery).forEach((key) => {  // Updated code
```